pypesto/objective/history.py

```diff
@@ -1315,9 +1350,20 @@ def update(
         self.history.update(x, sensi_orders, mode, result)
         self._update_vals(x, result)
 
-    def finalize(self):
-        """Finalize history."""
+    def finalize(self, message: str = None, exitflag: int = None):
+        """
+        Finalize history.
+
+        Parameters
+        ----------
+        message:
+            Optimizer message to be saved.
+        exitflag:
+            Optimizer exitflag to be saved.
+        """
         self.history.finalize()
+        if isinstance(self.history, Hdf5History):
+            self.history.finalize(message=message, exitflag=exitflag)
```

Comment:
this looks not clean -- the arguments should probably be passed to the history in any case without type checking.

Reply:
I was thinking about this as well, but that would need me to include those in the finalize of e.g. `MemoryHistory` which definitely does not need to save this in any way. Should I do it anyway? then probably in the `HistoryBase.finalize()`?